tests/common/schedulers/test_slurm_commands.py

```diff
@@ -83,82 +86,122 @@ def test_is_static_node(nodename, expected_is_static):
-                "######"
-                "NodeName=test-no-partition\n"
-                "NodeAddr=test-no-partition\n"
-                "NodeHostName=test-no-partition\n"
-                "State=IDLE+CLOUD+POWER\n"
-                # missing partitions
-                "######"
-            ),
+            "NodeName=multiple-st-c5xlarge-1\n"
+            "NodeAddr=172.31.10.155\n"
+            "NodeHostName=172-31-10-155\n"
+            "State=MIXED+CLOUD\n"
+            "Partitions=multiple\n"
+            "SlurmdStartTime=2023-01-23T17:57:07\n"
+            "######\n"
+            "NodeName=multiple-dy-c5xlarge-2\n"
+            "NodeAddr=172.31.7.218\n"
+            "NodeHostName=172-31-7-218\n"
+            "State=IDLE+CLOUD+POWER\n"
+            "Partitions=multiple\n"
+            "SlurmdStartTime=2023-01-23T17:57:07\n"
+            "######\n",
+            [
+                StaticNode(
+                    "multiple-st-c5xlarge-1",
+                    "172.31.10.155",
+                    "172-31-10-155",
+                    "MIXED+CLOUD",
+                    "multiple",
+                    slurmdstarttime=datetime(2023, 1, 23, 17, 57, 7).astimezone(tz=timezone.utc),
+                ),
+                DynamicNode(
+                    "multiple-dy-c5xlarge-2",
+                    "172.31.7.218",
+                    "172-31-7-218",
+                    "IDLE+CLOUD+POWER",
+                    "multiple",
+                    slurmdstarttime=datetime(2023, 1, 23, 17, 57, 7).astimezone(tz=timezone.utc),
+                ),
+            ],
+            False,
+        ),
+        (
+            "NodeName=multiple-dy-c5xlarge-3\n"
+            "NodeAddr=multiple-dy-c5xlarge-3\n"
+            "NodeHostName=multiple-dy-c5xlarge-3\n"
+            "State=IDLE+CLOUD+POWER\n"
+            "Partitions=multiple\n"
+            "Reason=some reason  \n"
+            "SlurmdStartTime=None\n"
+            "######\n",
             [
-                StaticNode("multiple-st-c5xlarge-1", "172.31.10.155", "172-31-10-155", "MIXED+CLOUD", "multiple"),
-                DynamicNode("multiple-dy-c5xlarge-2", "172.31.7.218", "172-31-7-218", "IDLE+CLOUD+POWER", "multiple"),
                 DynamicNode(
                     "multiple-dy-c5xlarge-3",
                     "multiple-dy-c5xlarge-3",
                     "multiple-dy-c5xlarge-3",
                     "IDLE+CLOUD+POWER",
                     "multiple",
-                    "some reason",
+                    "some reason  ",
+                    slurmdstarttime=None,
                 ),
+            ],
+            False,
+        ),
+        (
+            "NodeName=multiple-dy-c5xlarge-4\n"
+            "NodeAddr=multiple-dy-c5xlarge-4\n"
+            "NodeHostName=multiple-dy-c5xlarge-4\n"
+            "State=IDLE+CLOUD+POWER\n"
+            "Partitions=multiple,multiple2\n"
+            "Reason=(Code:InsufficientInstanceCapacity)Failure when resuming nodes \n"
```

Comment:
With the changes in code: `(Reason=.+) |(######)` to `(Reason=.*)|(######)` Looks like we are passing more information to Reasons, could you change this test case accordingly?

Reply:
with `scontrol show nodes` result:
`Reason=test [root@2023-01-31T21:24:55]`
Previously, we are using regex `(Reason=.+) |(######)`, with that expression, the match will be `Reason=test `
With your change, `(Reason=.*)|(######)`, the match will be `Reason=test [root@2023-01-31T21:24:55]`
Can we include that in this test case?

Reply:
Done.